pkg/agent/controller/packetcapture/packetcapture_controller_linux.go

```diff
@@ -0,0 +1,879 @@
+		}
+		return status, nil
+	}()
+	if err != nil {
+		return err
+	}
+	if err := c.patchStatusIfNeeded(pcName, status); err != nil {
+		return fmt.Errorf("error when patching status: %w", err)
+	}
+	return nil
+}
+
+func (c *Controller) patchStatusIfNeeded(name string, status string) error {
+	klog.InfoS("Patching PacketCapture CR Status if needed", "name", name, "status", status)
+	conditions := []crdv1alpha1.PacketCaptureCondition{}
+
+	if status == "RUNNING" {
+		conditions = append(conditions, crdv1alpha1.PacketCaptureCondition{
+			Type:               crdv1alpha1.PacketCaptureRunning,
+			Status:             metav1.ConditionStatus(v1.ConditionTrue),
+			LastTransitionTime: metav1.Now(),
+		})
+	} else if status == "WAITING" || status == "" {
+		conditions = append(conditions, crdv1alpha1.PacketCaptureCondition{
+			Type:               crdv1alpha1.PacketCapturePending,
+			Status:             metav1.ConditionStatus(v1.ConditionTrue),
+			LastTransitionTime: metav1.Now(),
+		})
+	} else {
+		return nil
+	}
+
+	toUpdate, getErr := c.packetCaptureLister.Get(name)
+	if getErr != nil {
+		return getErr
+	}
+	if retryErr := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		newCond := mergeConditions(toUpdate.Status.Conditions, conditions)
+		if conditionSliceEqualsIgnoreLastTransitionTime(toUpdate.Status.Conditions, newCond) {
+			return nil
+		}
+		toUpdate.Status.Conditions = newCond
+		klog.V(2).InfoS("Updating PacketCapture", "name", name, "status", toUpdate.Status)
+		_, updateErr := c.crdClient.CrdV1alpha1().PacketCaptures().UpdateStatus(context.TODO(), toUpdate, metav1.UpdateOptions{})
+		if updateErr != nil && apierrors.IsConflict(updateErr) {
+			var getErr error
+			if toUpdate, getErr = c.crdClient.CrdV1alpha1().PacketCaptures().Get(context.TODO(), name, metav1.GetOptions{}); getErr != nil {
+				return getErr
+			}
+		}
+		return updateErr
+	}); retryErr != nil {
+		return retryErr
+	}
+
+	return nil
+}
+
+func (c *Controller) getUploaderByProtocol(protocol StorageProtocolType) (ftp.Uploader, error) {
+	if protocol == sftpProtocol {
+		return c.sftpUploader, nil
+	}
+	return nil, fmt.Errorf("unsupported protocol %s", protocol)
+}
+
+func (c *Controller) generatePacketsPathForServer(name string) string {
+	return name + ".pcapng"
+}
+
+func getDefaultFileServerAuth() *crdv1alpha1.BundleServerAuthConfiguration {
+	return &crdv1alpha1.BundleServerAuthConfiguration{
+		AuthType: crdv1alpha1.BasicAuthentication,
+		AuthSecret: &v1.SecretReference{
+			Name:      fileServerAuthSecretName,
+			Namespace: env.GetAntreaNamespace(),
+		},
+	}
+}
+
+func (c *Controller) uploadPackets(pc *crdv1alpha1.PacketCapture, outputFile afero.File) error {
+	klog.V(2).InfoS("Uploading captured packets for PacketCapture", "name", pc.Name)
+	uploader, err := c.getUploaderByProtocol(sftpProtocol)
+	if err != nil {
+		return fmt.Errorf("failed to upload packets while getting uploader: %v", err)
+	}
+	authConfig := getDefaultFileServerAuth()
```

Comment:
we are not using configurable authentication like for SupportBundle?

Reply:
in the origin design and the previous MR, it was. Then I think Quan suggests to use a dedicated secret.